### PR TITLE
NIFI-10538: UI updates for making RegistryClient an extension point

### PR DESCRIPTION
**Pull request by sardell**
This PR contains UI changes around Add/Edit Registry dialogs as a result of changes in https://github.com/apache/nifi/pull/6433.

A quick summary of the work in this PR: 
* Previously, we utilized one dialog for both adding and editing a registry. Because we need to support adding/editing/deleting properties, the two dialogs have much less in common now. I've separated the dialogs into two distinct .jsp files.
* I removed the url input in the Add Registry dialog and added a combo dropdown for class type selection.
* ~I added tabs to the Edit Registry dialog. One of these tabs contains a table for adding/editing properties. I'm currently waiting to see if #6433 will support dynamic properties. If it doesn't, I'll need to update this work to conditionally show/hide the add property button. It currently does not work in this PR.~ This PR now supports dynamic properties
* ~This PR contains all the changes from #6433. I'm unable to start NiFi with latest work on that branch due to some java errors, which @simonbence is aware of. I'm going to open this PR as a draft until that's resolved.~ This is now resolved.

# Summary

[NIFI-10538](https://issues.apache.org/jira/browse/NIFI-10538)

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [ ] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [ ] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [ ] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [ ] Pull Request based on current revision of the `main` branch
- [ ] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [ ] Build completed using `mvn clean install -P contrib-check`
  - [ ] JDK 8
  - [ ] JDK 11
  - [ ] JDK 17

### Licensing

- [ ] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [ ] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [ ] Documentation formatting appears as expected in rendered files

### CODE REVIEW

**Comment by mcgilman** [comment]
Will review...

**Review by mcgilman — CHANGES_REQUESTED** [review]
Nice work @sardell! I've left a few comments below.

Also, with the introduction of the Controller Service references we need to update the CS dialogs to ensure those references are shown.

Screen Shot 2022-09-22 at 10 04 12 AM

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/WEB-INF/partials/canvas/registry-configuration-dialog.jsp [hunk 1]
```diff
@@ -15,26 +15,45 @@
...
+                    <div class="setting">
+                        <div class="setting-name">Id</div>
+                        <div class="setting-field">
+                            <span id="registry-id-config"></span>
+                        </div>
+                    </div>
+                    <div class="setting">
+                        <div class="setting-name">Name</div>
+                        <div class="setting-field">
+                            <span id="registry-name-config"></span>
```
[the remark this answers is not in the document]
Comment (sardell) [reply nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/WEB-INF/partials/canvas/registry-configuration-dialog.jsp:36]: That was my misunderstanding. @simonbence just pointed this out to me as well.

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-controller-service.js [hunk 2]
```diff
@@ -257,10 +257,12 @@
                 // reload
                 nfParameterProvider.reload(reference.id);
 
+            } else if (reference.referenceType === 'FlowRegistryClient') {
```
Comment (mcgilman) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-controller-service.js:260]: I think something may have been mixed up during a rebase here.

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-flow-version.js [hunk 3]
```diff
@@ -206,7 +206,8 @@
                     registries.push({
                         text: registry.name,
                         value: registry.id,
-                        description: nfCommon.escapeHtml(registry.description)
+                        description: nfCommon.escapeHtml(registry.description),
+                        disabled: registry.validationStatus !== 'VALID' // TODO: verify this logic is correct
```
Comment (mcgilman) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-flow-version.js:210]: This looks correct but we should check with @simonbence or @markap14 who are a little more familiar with the backend effort [1].

[1] #6433

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-controller-service.js [hunk 4]
```diff
@@ -402,6 +404,36 @@
         });
     };
 
+    var updateValidationErrors = function (validationErrorIcon, referencingComponent) {
```
Comment (mcgilman) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-controller-service.js:407]: This function appears to be a duplicate. Likely another mixed up with the rebase. Also, there's another condition that may have been missed. Search for 'ParameterProvider' to find other instances.

**Review by mcgilman — COMMENTED** [review]
Thanks @sardell. Couple minor items belows.

**Review by mcgilman — APPROVED** [review]
Thanks @sardell. Nice work! Will merge once CI completes.

**Pull request merged**